drivers/soundwire/mipi_disco.c

```diff
@@ -330,8 +331,11 @@ int sdw_slave_read_prop(struct sdw_slave *slave)
 	device_property_read_u32(dev, "mipi-sdw-clockstopprepare-timeout",
 				 &prop->clk_stop_timeout);
 
-	device_property_read_u32(dev, "mipi-sdw-slave-channelprepare-timeout",
-				 &prop->ch_prep_timeout);
+	ret = device_property_read_u32(dev, "mipi-sdw-peripheral-channelprepare-timeout",
```

Comment:
Yeah, not the first time and most likely not the last we will see similar fallbacks. Perhaps you can extend the generic device property API to support this? Like you supply the list of the possible properties in the order of high to low priority
```
static inline int device_property_read_u32_one_of(..., const char * const *properties, u32 *value)
{
  const char *p;
  int ret;

  for (p = *properties; p; p++) {
    ret = ...;
    if (ret != ...) // code for no property found
      return ret;
  }

  return ...; // same code as per above
}
```
either as inliners or macros (with a type passed through `_Generic()`)?

Reply:
Humm, not sure. This would work when the change is only a renaming, but the parameters and meaning mean the same. But when the value changes between a count and a bitmask, you need to know which of the two properties was found to deal with error handling.

Reply:
I see what you mean. I even have a real case as you described, but still we may return an index of the property that were found and parsed: < 0 — an error, >= 0 — an index in property names area, from which one may decide how to interpret the result.